rollup.config.js

```diff
@@ -2,16 +2,68 @@ import babel from 'rollup-plugin-babel';
 import localResolve from 'rollup-plugin-local-resolve';
 import json from 'rollup-plugin-json';
 
-export default {
-  format: 'cjs',
-  plugins: [
-    localResolve(),
-    json(),
-    babel({
-      babelrc: false,
-      presets: [['es2015', { modules: false, loose: true }]],
-      plugins: ['transform-decorators-legacy', 'transform-class-properties', 'transform-runtime'],
-      runtimeHelpers: true
-    })
-  ]
-};
+function createConfig(filename, browserlist, suffix = '') {
+  return {
+    input: `src/${filename}`,
+    output: {
+      file: `${filename}${suffix}.js`,
+      format: 'cjs',
+      sourcemap: true
+    },
+    external: [
+      'restructure',
+      'tiny-inflate',
+      'brotli/decompress',
+      'unicode-properties',
+      'clone',
+      'deep-equal',
+      'unicode-trie',
+      'dfa',
+      'restructure/src/utils'
+    ],
+    plugins: [
+      localResolve(),
+      json(),
+      babel({
+        presets: [
+          [
+            '@babel/preset-env', 
+            {
+              modules: false,
+              targets: {
+                browsers: browserlist
+              }
+            }
+          ]
+        ],
+        plugins: [
+          [
+            '@babel/plugin-proposal-decorators',
+            {
+              legacy: true
+            }
+          ],
+          '@babel/plugin-proposal-class-properties'
+        ]
+      })
+    ]
+  }
+}
+
+const modernBrowsers = [
```

Comment:
probably worth using https://github.com/browserslist/browserslist instead?

Reply:
This can be done is a followup PR.

Reply:
given how much further into the future things are now, it's probably worth updating this to https://github.com/browserslist/browserslist rather than in followup at this point. Just in case it takes another year before it actually lands =(